regressiontests/regression_test.go

```diff
@@ -0,0 +1,348 @@
+			}
+		}()
+
+	default:
+		t.Fatalf("Unsupported database engine: %s", dbEngine)
+	}
+
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	defer func() {
+		t.Log("Killing application process...")
+		t.Logf("pid: %d", cmd.Process.Pid)
+		if err := cmd.Process.Kill(); err != nil {
+			t.Fatalf("Failed to kill application process: %v", err)
+		}
+	}()
+
+	if err := cmd.Start(); err != nil {
+		t.Fatalf("Failed to start the application: %v", err)
+	}
+
+	appExitSignal := make(chan error, 1)
+	go func() {
+		err := cmd.Wait()
+		appExitSignal <- err
+	}()
+
+out:
+	for {
+		select {
+		case err := <-appExitSignal:
+			if err != nil {
+				t.Fatalf("Application exited unexpectedly: %v", err)
+			}
+		case <-timeoutTimer.C:
+			t.Fatalf("Test timed out after 2 minutes without passing all checks.")
+		case <-ticker.C:
+			resp, err := http.Get("http://localhost:8080/status")
+			if err != nil {
+				t.Logf("Failed to make HTTP request to the application: %v - next attempt in 10s", err)
+				continue
+			}
+			defer func(Body io.ReadCloser) {
+				err := Body.Close()
+				if err != nil {
+					t.Logf("Couldn't close http status request body %v", err)
+				}
+			}(resp.Body)
+
+			if resp.StatusCode != 200 {
+				t.Logf("Expected status code 200, got %d - next attempt in 10s", resp.StatusCode)
+				continue
+			}
+
+			t.Log("HTTP server is up and running")
+
+			wocTipHeight, err := fetchExternalForkHeight(whatsonchainAPIURL)
+			if err != nil {
+				t.Logf("Failed to fetch external fork height: %v", err)
+				continue
+			}
+
+			localTipHeight, err := fetchLocalTip()
+			if err != nil {
+				t.Logf("Failed to fetch local tip height: %v", err)
+				continue
+			}
+
+			if localTipHeight < wocTipHeight {
+				t.Logf("Couldn't sync to proper tip of chain - next attempt in 10s: %d < %d", localTipHeight, wocTipHeight)
+				continue
+			} else {
+				t.Logf("Synced to tip of chain: %d", localTipHeight)
+				break out
+			}
+		}
+	}
+
+	t.Log("Comparing synced data with known checkpoints...")
+
+	verifyHeaders(t)
+
+	t.Log("Headers checkpoint comparison passed successfully 🎉")
+
+	t.Log("Verifying merkle roots...")
+
+	verifyMerkleRoots(t, merkleRootFixtures)
+
+	t.Log("Merkle roots verification passed successfully 🎉")
+}
+
+func startPostgresContainer(ctx context.Context, t *testing.T) (testcontainers.Container, string) {
+	req := testcontainers.ContainerRequest{
+		Image:        "postgres:latest",
+		ExposedPorts: []string{"5432/tcp"},
+		Env: map[string]string{
+			"POSTGRES_USER":     "testuser",
+			"POSTGRES_PASSWORD": "test",
+			"POSTGRES_DB":       "bhs",
+		},
+		WaitingFor: wait.ForListeningPort("5432/tcp"),
+	}
+	postgresContainer, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	if err != nil {
+		t.Fatalf("Could not start postgres container: %v", err)
+	}
+
+	time.Sleep(10 * time.Second) // Wait for the container to be ready
```

Comment:
Is it necessary even though the option `WaitingFor: wait.ForListeningPort("5432/tcp"),` is utilized.

Reply:
removed this option